include/irr/asset/CTexturePacker.h

```diff
@@ -81,57 +80,32 @@ class ITexturePacker
         }
         uint32_t x() const { return addr&X_MASK; }
         uint32_t y() const { return (addr>>(ADDR_LAYER_SHIFT>>1))&X_MASK; }
-        uint32_t layer() const { return addr>>ADDR_LAYER_SHIFT; }
+        uint32_t layer() const { return (addr>>ADDR_LAYER_SHIFT)&0xffffu; }
+
+        bool isValid() const { return (addr&0xffffu)==invalid_addr; }
+        bool hasMipTailAddr() const { return addr&(0xffffu<<16); }
+        STexOffset getMipTailAddr() const { return addr>>16; }
 
+        //upper 16 bits are used for address of mip-tail page
         uint32_t addr;
     };
 
+    //TODO will work only with _tilesPerDim=32, for different values ADDR_LAYER_SHIFT must be adjusted as well to correctly retrieve 2d+layer addr from morton+layer
```

Comment:
Make the `x()`, `y()` and `layer()` functions of `STexOffset` be handled via the texture packer